Corrected.php

```diff
@@ -3,35 +3,110 @@
 /**
  * Some namespace, some libraries, some annotations.
  */
-class ShiftCenterRestResource extends ResourceBase {
-
-  /**
-   * Some code.
-   */
-
-  /**
-   * Get if user has access to this shift center.
-   *
-   * @param integer $uid
-   *   User id.
-   * @param integer $id
-   *   Shift center id.
-   *
-   * @return bool
-   *   True if user has access.
-   */
-  private function getAccess($uid, $id) {
+class ShiftCenterRestResource extends ResourceBase
+{
+
     /**
      * Some code.
      */
 
-    $access = TRUE;
+    /**
+     * Get if user has access to this shift center.
+     *
+     * @param integer $uid
+     *   User id.
+     * @param integer $id
+     *   Shift center id.
+     *
+     * @return bool
+     *   True if user has access.
+     */
+    private function getAccess($uid, $id)
+    {
+        $key = $this->accessKeyGenerator($uid, $uid);
+
+        return $this->hasDrupalAccess($key) || $this->hasCachedAccess($key) ??
+               $this->checkActionPermissions($uid, $uid);
+    }
+
+    /**
+     * @param int $uid
+     * @param int $id
+     *
+     * @return bool
+     */
+    private function checkActionPermissions(int $uid, int $id) : bool
+    {
+        $hasAccess = $this->shift_access->getPositionSameGroup($uid, $id) &&
```

Comment:
Should be abstracted to reusable components with configuration. 
I don't work with Drupal but it seems that `getAccess` responsibility maybe shouldn't be in `ShiftCenterRestResource` but in an `AccessProvider` component. No way I would have gone for this :)